diff --git a/CMakeLists.txt b/CMakeLists.txt
index 417789a..39968f1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@ set(HPX_KOKKOS_VERSION_STRING "${HPX_KOKKOS_VERSION_MAJOR}.${HPX_KOKKOS_VERSION_
 find_package(HPX 1.8.1 REQUIRED)
 # TODO: The version requirement needs to be bumped once
 # https://github.com/kokkos/kokkos/pull/5628 is merged and released.
-find_package(Kokkos 3.7.99 REQUIRED)
+find_package(Kokkos 3.6.01 REQUIRED)
 
 if(Kokkos_ENABLE_CUDA)
   kokkos_check(OPTIONS CUDA_LAMBDA CUDA_LAMBDA)
